adapters/33across/33across.go

```diff
@@ -38,39 +55,59 @@ func (a *TtxAdapter) MakeRequests(request *openrtb.BidRequest, reqInfo *adapters
 	var errs []error
 	var adapterRequests []*adapters.RequestData
 
-	adapterReq, errors := a.makeRequest(request)
-	if adapterReq != nil {
-		adapterRequests = append(adapterRequests, adapterReq)
+	// Break up multi-imp request into multiple external requests since we don't
+	// support SRA in our exchange server
+	for i := 0; i < len(request.Imp); i++ {
+		// makeRequest will either return an adapterRequest or err
+		if adapterReq, err := a.makeRequest(request, request.Imp[i]); err == nil {
+			adapterRequests = append(adapterRequests, adapterReq)
+		} else {
+			errs = append(errs, err)
+		}
 	}
-	errs = append(errs, errors...)
 
-	return adapterRequests, errors
+	return adapterRequests, errs
 }
 
-// Update the request object to include custom value
-// site.id
-func (a *TtxAdapter) makeRequest(request *openrtb.BidRequest) (*adapters.RequestData, []error) {
-	var errs []error
+func (a *TtxAdapter) makeRequest(request *openrtb.BidRequest, imp openrtb.Imp) (*adapters.RequestData, error) {
+	var imps []openrtb.Imp
 
-	// Make a copy as we don't want to change the original request
 	reqCopy := *request
-	if err := preprocess(&reqCopy); err != nil {
-		errs = append(errs, err)
+	impCopy, err := makeImps(imp)
```

Comment:
I'd suggest extracting this logic related to `reqExt` into it's own function and calling it before calling `makeRequest` for each imp because right now this code is executed for each imp but that's not really needed as the `reqExt` is gonna be the same and so you don't need to unmarshal and marshal it for each imp.

Once you do that, you can then only pass the `reqExt` and the `imp` under consideration to the `makeRequest` function.

JSON marshaling and unmarshaling is an expensive operation and we wanna make sure we avoid doing that multiple times unless it is absolutely needed.

Reply:
Done.